woocommerce/woocommerce-android
# [Hack Week] ApiFaker: Refactor the developer screen viewmodel and fragment

### Description
As part of my Hack Week, I'm bringing back the ApiFaker project paqN3M-SI-p2, and I'm splitting the feature into multiple PRs.

This first PR is just a preparation, it refactors the "Developer Options" code:
- It refactors the ViewModel to improve the readability of the code and to make adding new buttons easier.
- It converts the screen to use Compose instead.

I intentionally didn't make any big changes to the design, and I didn't work on adding any new unit tests.

**Note:** Reviewing commit by commit would be easier, as Github is marking some files as new while they were just moved to a different package.

### Steps to reproduce
1. Use a debug build.
2. Open the settings screen.
3. Tap on "Developer Options".

### Testing information
- Confirm the screen works as expected.
- Enable the Simulated card reader, and test connection to it from the payment screen.

### The tests that have been performed
^

### Images/gif
| Before | After |
| --- | --- |
| ![Screenshot_20241205_100651](https://github.com/user-attachments/assets/0ed8fe5d-92a9-4784-8869-ca7ac7ec0557) | ![Screenshot_20241205_101115](https://github.com/user-attachments/assets/233980b5-03c9-43d2-aba9-45403c677219)| 



## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
